### PR TITLE
MOTOR-969 Support for Range Indexes

### PR DESCRIPTION
**Pull request by juliusgeo**

### CODE REVIEW

File: test/tornado_tests/test_motor_collection.py [hunk 1]
```diff
@@ -283,6 +285,57 @@ async def test_async_create_encrypted_collection(self):
             self.addCleanup(self.db.drop_collection, "testing1", encrypted_fields=ef)
             self.assertEqual(exc.exception.code, 121)
 
+    @gen_test
+    async def test_async_encrypt_expression(self):
```
Comment (blink1073) [inline test/tornado_tests/test_motor_collection.py:289]: This test and the `test_async_create_encrypted_collection` should also be in `tests/asyncio_tests/test_asyncio_collection.py`

File: motor/metaprogramming.py [hunk 2]
```diff
@@ -149,6 +149,8 @@ def __init__(self, attr_name, doc=None):
 
     def create_attribute(self, cls, attr_name):
         name = self.attr_name or attr_name
+        if name == "encrypt_expression":
+            return
```
Comment (blink1073) [inline motor/metaprogramming.py:153]: You could instead add `None` as the last arg to `getattr()` and return if it is None.

Comment (juliusgeo) [reply motor/metaprogramming.py:153]: My concern with doing it that way is that we might forget to get rid of this workaround later. In general I think we do want to raise an error in this case (for example `Sync("method_that_is_mispelled"`), so I want it to be as specific as possible.

Comment (blink1073) [reply motor/metaprogramming.py:153]: This unconditionally gets rid of the attribute though right?  I think we need to first check if it exists, and then check if it is in an expected list of ones that may not exist yet.

Comment (juliusgeo) [reply motor/metaprogramming.py]: Ah, that is a very good point. Done.

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]

File: test/tornado_tests/test_motor_collection.py (hunk 1 above)
Comment (juliusgeo) [reply test/tornado_tests/test_motor_collection.py:293]: Done.

File: motor/core.py [hunk 3]
```diff
@@ -1979,6 +1979,7 @@ class AgnosticClientEncryption(AgnosticBase):
     add_key_alt_name = AsyncCommand()
     get_key_by_alt_name = AsyncCommand()
     remove_key_alt_name = AsyncCommand()
+    encrypt_expression = AsyncCommand()
```
[the remark this answers is not in the document]
Comment (juliusgeo) [reply motor/core.py]: That makes sense. Done.

**Review by ShaneHarvey — APPROVED** [review]

**Review by blink1073 — APPROVED** [review]

File: test/asyncio_tests/test_asyncio_collection.py [hunk 4]
```diff
@@ -260,6 +266,79 @@ def test_sub_collection(self):
...
+            KMS_PROVIDERS, "keyvault.datakeys", c, CodecOptions()
+        ) as client_encryption:
+            coll, ef = await client_encryption.create_encrypted_collection(
+                database=self.db,
+                name="testing1",
+                encrypted_fields={"fields": [{"path": "ssn", "bsonType": "string", "keyId": None}]},
+                kms_provider="local",
+            )
+            with self.assertRaises(pymongo.errors.WriteError) as exc:
+                coll.insert_one({"ssn": "123-45-6789"})
```
Comment (ShaneHarvey) [inline test/asyncio_tests/test_asyncio_collection.py:287]: How is this working without the await? What is the type of `coll` here?

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]

File: test/asyncio_tests/test_asyncio_collection.py (hunk 4 above)
Comment (juliusgeo) [reply test/asyncio_tests/test_asyncio_collection.py]: This should be fixed now.

Comment (ShaneHarvey) [reply test/asyncio_tests/test_asyncio_collection.py]: How was it passing before?

Comment (juliusgeo) [reply test/asyncio_tests/test_asyncio_collection.py]: Before there were no awaits, so the function calls were returning immediately and not raising any exceptions.

**Comment by blink1073** [comment]
Can you pull from master to pick up the `tox` fix from https://github.com/mongodb/motor/pull/194?

**Comment by juliusgeo** [comment]
@blink1073 I appear to already have that commit:
 ```
(motor3.9) ➜  motor git:(MOTOR-969) git log | grep 194
    MOTOR-1095 Make Rust Compiler available for Cryptography build (#194)
```

**Comment by blink1073** [comment]
Hmm, then we might need to use `ignore_basepython_conflict = true`, https://tox.wiki/en/latest/config.html#python-language-core-options

File: test/asyncio_tests/test_asyncio_collection.py (hunk 4 above)
Comment (ShaneHarvey) [reply test/asyncio_tests/test_asyncio_collection.py]: Apparently this was failing before.

**Review by ShaneHarvey — APPROVED** [review]

**Comment by juliusgeo** [comment]
@blink1073 I opened https://jira.mongodb.org/browse/MOTOR-1104 to track that work.

**Review by blink1073 — APPROVED** [review]

**Pull request merged**